pesq/cypesq.pyx

```diff
@@ -49,14 +72,31 @@ cdef extern from "pesqio.h":
 cdef extern from "pesqmain.h":
     cdef void pesq_measure(SIGNAL_INFO * ref_info, SIGNAL_INFO * deg_info, ERROR_INFO * err_info, long * Error_Flag, char ** Error_Type)
 
+# Well, Globals are evil, but this is just storing the last message. 
+# It should be ok-ish.
+cdef char* last_error_message = "unknown";
+
+cdef void set_last_error_message(char* msg):
+    global last_error_message
+    last_error_message = msg
```

Comment:
Is this operation thread safe?
When someone plans to release the GIL and want to make the code thread safe, this may produce a problem, because it is part of the API.

Reply:
Definitely not thread safe, but I'm not sure it is important to be. It's there more for convenience as the message it provides does not really gives you more information than the return codes, which I think are thread safe. Also, I've seen some globals while implementing this patch, so I'd say @boeddeker is correct and we should assume the whole C backend isn't threadsafe.

Reply:
I personally would implement a lookup table to convert the error code to the actual message.
The critical point here is,  from the user side this is not thread safe.

For the case there is someone planning to make the complete codebase thread safe, this function must be deprecated.
For all the other non thread safe positions it is not a problem, because you can change it, without breaking user code.

Reply:
The messages comes from deep within the API, and I didn't want to return 2 values in pesq_measure. I'm trying not to change too much stuff outside the surface-level API with this PR. In case someone wants to make this thread-safe, it should be trivial to do so.

Reply:
So, turns out most of the messages were not very helpful anyway, so I just picked the best ones and added them to a lookup table as suggested.